gwas_sumstats_tools/gen_meta.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from rich import print
+from rich.progress import Progress, SpinnerColumn, TextColumn
+
+from gwas_sumstats_tools.interfaces.metadata import (
+    MetadataClient,
+    metadata_dict_from_gwas_cat,
+    get_file_metadata,
+)
+from gwas_sumstats_tools.utils import (
+    parse_accession_id,
+    append_to_path,
+    exit_if_no_data,
+)
+
+class Gen_meta:
+    def __init__(
+        self,
+        data_infile: Path,
+        metadata_infile: Path = None,
+        metadata_outfile: Path = None,
+        format_data: bool = False,
+    ) -> None:
+        self.format_data = format_data
+        self.data_infile = Path(data_infile)
+        self.metadata_outfile = Path(
+            self._set_metadata_outfile_name()
+            if not metadata_outfile
+            else metadata_outfile
+        )
+        self.meta = MetadataClient(
+            in_file=metadata_infile, out_file=self.metadata_outfile
+        )
+
+    def _set_metadata_outfile_name(self) -> str:
+        #return append_to_path(self.data_outfile, "-meta.yaml")
+        return append_to_path(self.data_infile, "-meta.yaml")
```

Comment:
Method name is `_set_metadata_outfile_name` but it uses `self.data_infile`. Is this a typo?

Reply:
Thank you for your comments. data_infle is is GCST1234.tsv, the output meta_yaml file is GCST1234.tsv-meta.yaml. So it is no problem for self.data_infile here.